frontend/src/features/public-form/utils/inputTransformation.ts

```diff
@@ -62,41 +66,55 @@ const transformToVerifiableOutput = <
   input: unknown,
 ): VerifiableAnswerOutput<F> => {
   if (
-    !validateVerifiableInput(input) ||
-    (schema.isVerifiable && !input.signature)
+    input !== undefined &&
+    (!validateVerifiableInput(input) ||
+      (schema.isVerifiable && !input.signature))
   ) {
     throw new InputValidationError(schema._id, input)
   }
   return {
     ...pickBaseOutputFromSchema(schema),
-    answer: input.value,
-    signature: input.signature,
+    answer: input?.value ?? '',
+    signature: input?.signature,
   }
 }
 
 const transformToSingleAnswerOutput = <F extends FormFieldDto>(
   schema: F,
   input: unknown,
 ): SingleAnswerOutput<F> => {
-  if (!validateSingleAnswerInput(input)) {
+  if (input !== undefined && !validateSingleAnswerInput(input)) {
```

Comment:
why do you need the undefined check when `validateSingleAnswerInput` already checked for `typeof input === 'string'`?

Reply:
same for all the other `undefined` checks